R/shared-methods.R

```diff
@@ -96,13 +96,13 @@ initialise_dataclass <- function(self, level = "1", regions,
     self$target_regions <- regions
   }
 
-  if (!is.null(self$level_data_urls[[self$level]])) {
+  if (length(self$common_data_urls) > 0) {
     self$data_urls <- merge(
       self$common_data_urls,
       self$level_data_urls[[self$level]]
     )
   } else {
-    self$data_urls <- self$common_data_urls
+    self$data_urls <- self$level_data_urls[[self$level]]
```

Comment:
Once merge in we would ideally have some tests to check this logic works as we think it does.